ProfilerGUI/Source/Runner/ProfilerRunner.cs

```diff
@@ -59,16 +55,54 @@ public void RunAsynchronously()
             {
                 logger.Error(e, "Failed to launch application {appPath}", configuration.TargetApplicationPath);
             }
-        }
-
-        private string LocateProfilerDll()
-        {
-            string profilerDll = ProfilerConstants.ProfilerDll64Bit;
-            if (configuration.ApplicationType == EApplicationType.Type32Bit)
+        }
+
+        private static List<(string, string)> DetermineEnvironmentVariables(ProfilerConfiguration configuration)
+        {
+            List<(string, string)> environmentVariables;
+            if (configuration.ApplicationType == EApplicationType.TypeCORE)
+            {
+                environmentVariables = new List<(string, string)>()
+                {
+                    (ProfilerConstants.CoreProfilerIdEnvironmentVariable, ProfilerConstants.ProfilerGuid),
+                    (ProfilerConstants.CoreProfilerPath32EnvironmentVariable, GetProfilerDllPath(ProfilerConstants.ProfilerDll32Bit)),
+                    (ProfilerConstants.CoreProfilerPath64EnvironmentVariable, GetProfilerDllPath(ProfilerConstants.ProfilerDll64Bit)),
+                    (ProfilerConstants.CoreEnableProfilingEnvironmentVariable, "1"),
+                };
+            }
+            else
+            {
+                string profilerDll = ProfilerConstants.ProfilerDll64Bit;
+                if (configuration.ApplicationType == EApplicationType.Type32Bit)
+                {
+                    profilerDll = ProfilerConstants.ProfilerDll32Bit;
+                }
+                
+                environmentVariables = new List<(string, string)>()
+                {
+                    (ProfilerConstants.ProfilerIdEnvironmentVariable, ProfilerConstants.ProfilerGuid),
+                    (ProfilerConstants.ProfilerPathEnvironmentVariable, GetProfilerDllPath(profilerDll)),
+                    (ProfilerConstants.EnableProfilingEnvironmentVariable, "1"),
+                };
+            }
+            
+            environmentVariables.AddRange(new List<(string, string)>()
+            {
+                (ProfilerConstants.TargetDirectoryEnvironmentVariable, configuration.TraceTargetFolder),
+                (ProfilerConstants.LightModeEnvironmentVariable, "1")
+            });
+
+            return environmentVariables;
+        }
+
+        private static string GetProfilerDllPath(string profilerDll)
+        {
+            var path = Directory.GetParent(Environment.CurrentDirectory) + @"\" + profilerDll;
+            if (!File.Exists(path))
             {
-                profilerDll = ProfilerConstants.ProfilerDll32Bit;
-            }
-            return Directory.GetParent(Environment.CurrentDirectory) + @"\" + profilerDll;
-        }
+                throw new ArgumentException(string.Format("The profiler DLL was not found at \"{0}\". Cannot profile the application", path));
```

Comment:
I'm not 100% happy with this string formatting, but exceptions can't do it for us like the logger can and I wanted to have the explicit `logger.Error` in `RunAsynchronously()`, to make it explicit there that we report the error before aborting the run.

Reply:
You could do string interpolation `@"foo {path} bar"`